lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,394 @@
+                module.fail_json(msg=to_native(exc))
+
+            if self.ca_certificates:
+                ca_certs = [crypto_utils.load_certificate(ca_cert) for ca_cert
+                            in self.ca_certificates]
+                self.pkcs12.set_ca_certificates(ca_certs)
+
+            if self.cert_path:
+                self.pkcs12.set_certificate(crypto_utils.load_certificate(
+                                            self.cert_path))
+
+            if self.friendly_name:
+                self.pkcs12.set_friendlyname(self.friendly_name)
+
+            if self.privatekey_path:
+                self.pkcs12.set_privatekey(crypto_utils.load_privatekey(
+                                           self.privatekey_path,
+                                           self.privatekey_passphrase)
+                                           )
+
+            try:
+                with open(self.path, 'wb', this_mode) as archive:
+                    archive.write(
+                        self.pkcs12.export(
+                            self.passphrase,
+                            self.iter_size,
+                            self.maciter_size
+                        )
+                    )
+                module.set_mode_if_different(self.path, this_mode, False)
+                self.changed = True
+            except (IOError, OSError) as exc:
+                self.remove()
+                raise PkcsError(exc)
+
+        file_args = module.load_file_common_arguments(module.params)
+        if module.set_fs_attributes_if_different(file_args, False):
+            module.set_mode_if_different(self.path, this_mode, False)
+            self.changed = True
+
+    def parse(self, module):
+        ''' Read PKCS#12 file. '''
+
+        this_mode = module.params['mode']
+        if not this_mode:
+            this_mode = int('0400', 8)
+
+        if not self.check(module, perms_required=False) or module.params['force']:
+            try:
+                self.remove()
+
+                p12 = crypto.load_pkcs12(open(self.src, 'rb').read(),
+                                         passphrase=self.passphrase)
+                pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM,
+                                              p12.get_privatekey())
+                crt = crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                              p12.get_certificate())
+
+                with open(self.path, 'wb') as content:
+                    content.write("%s%s" % (pkey, crt))
+                module.set_mode_if_different(self.path, this_mode, False)
+                self.changed = True
+            except IOError as exc:
+                self.remove()
+                raise PkcsError(exc)
+
+        file_args = module.load_file_common_arguments(module.params)
+        if module.set_fs_attributes_if_different(file_args, False):
+            module.set_mode_if_different(self.path, this_mode, False)
```

Comment:
No need to call explicitly this, it is already called in the `module.set_fs_attributes_if_different()` call [1]

[1] https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1454-L1456